TheWitcher.Business/HeroService.cs

```diff
@@ -0,0 +1,239 @@
+﻿using System;
+using System.Linq;
+using DataAccessLayer.Core;
+using DataAccessLayer.Interfaces;
+using DataAccessLayer.Realization;
+
+
+namespace TheWitcher.Business
+{
+    public class HeroService
+    {
+        private IRepository<Heroes> _heroRepository;
+        private IRepository<Quest> _questRepository;
+        private IRepository<HeroInQuest> _heroInQuestRepository;
+        private IRepository<Clothes> _clothesRepository;
+        private IRepository<HeroClothes> _heroClothesRepository;
+        private IRepository<Weapons> _weaponsRepository;
+        private IRepository<HeroWeapon> _heroWeaponRepository;
+        private readonly UnitOfWork _unitOfWork;
+        public HeroService(UnitOfWork unitOfWork)
+        {
+            _heroRepository = unitOfWork.Hero;
+            _questRepository = unitOfWork.Quest;
+            _heroInQuestRepository = unitOfWork.HeroInQuest;
+            _clothesRepository = unitOfWork.Clothes;
+            _heroClothesRepository = unitOfWork.HeroClothes;
+            _weaponsRepository = unitOfWork.Weapon;
+            _heroWeaponRepository = unitOfWork.HeroWeapon;
+            _unitOfWork = unitOfWork;
+        }
+        private int CountPowerOfHero(int id)
+        {
+            int totalPower = 0;
+            var hero = _heroRepository.GetItem(id);
+            int clothesPower = hero.HeroClothes.Sum(x => x.Clothes.CombatPower).GetValueOrDefault();
+            totalPower += clothesPower;
+            int weaponsPower = hero.HeroWeapon.Sum(x => x.Weapons.CombatPower).GetValueOrDefault();
+            totalPower += weaponsPower;
+            return totalPower;
+        }
+        private bool ProcessCoefficient(Heroes hero, Quest quest, int heroPower)
+        {
+            try
+            {
+                int coefficient = heroPower / quest.Complexity.Value;
+                quest.Award = hero.HeroLevel / coefficient * 150;
+                double questTimeInSeconds = quest.LeadTime.Value.Seconds + quest.LeadTime.Value.Minutes * 60;
+                int newQuestTimeInSeconds = Convert.ToInt32(questTimeInSeconds / (hero.HeroLevel.Value * coefficient));
+                TimeSpan addingTimeSpan = TimeSpan.FromSeconds(newQuestTimeInSeconds);
+                quest.LeadTime = addingTimeSpan;
+                hero.ReleaseDate = DateTime.Now;
+                hero.ReleaseDate.Value.AddSeconds(newQuestTimeInSeconds);
+                return true;
+            }
+            catch(Exception e)
```

Comment:
It will be better to throw exception and handle it outside, so you can display some meaningful error message.
+ when you don't use exception object, use just catch (Exception)
+ try to handle specific exceptions. Because Exception will handle ALL type of exceptions, and you will not be able to identify what went wrong